lib/forwardable.rb

```diff
@@ -132,11 +132,7 @@ class << self
   #
   def instance_delegate(hash)
     hash.each do |methods, accessor|
-      if methods.respond_to?(:each)
```

Comment:
@ioquatix I have created two commits with implementation of your proposition in discussion https://github.com/ruby/forwardable/pull/1#discussion_r283163464
I have learned better current code base and and known, that each module(Forwardable, SingleForwardable) really has method, which implements processing of methods array, now, we are free from each or for, and also this replacing saves from redefining system methods:
Previously, method  _instance_delegate_ could delegate _system object methods_ like '__id__' or '__send__', but currently it won't happen because def_instance_delegators deletes system method from array.
Code example:
```ruby
# frozen_string_literal: true

require 'forwardable'

class Owner
  extend Forwardable

  attr_reader :object

  delegate __id__: :object
  delegate __send__: :object

  def initialize(object:)
    @object = object
  end

  def to_s
    'Instance of Owner'
  end
end

class Thing
  def to_s
    'Instance of Thing'
  end
end

thing = Thing.new

owner = Owner.new(object: thing)
p owner.__id__ == thing.__id__

puts owner.__send__(:to_s) # => 'Instance of Thing', but it's owner
```
Case above isn't really danger situation, it's just my journey and thinks